presto-main/src/main/java/com/facebook/presto/operator/aggregation/estimatendv/NDVEstimatorState.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.estimatendv;
+
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.spi.function.AccumulatorState;
+import com.facebook.presto.spi.function.AccumulatorStateMetadata;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+
+@AccumulatorStateMetadata(
+        stateSerializerClass = NDVEstimatorStateSerializer.class,
+        stateFactoryClass = NDVEstimatorStateFactory.class)
+public class NDVEstimatorState
+        implements AccumulatorState
+{
+    private Map<Long, Long> freqDict = new HashMap<>();
+
+    public NDVEstimatorState() {}
+
+    @Override
+    public long getEstimatedSize()
+    {
+        return 0;
+    }
+
+    public Map<Long, Long> getFreqDict()
+    {
+        return freqDict;
+    }
+
+    public void setFreqDict(Map<Long, Long> frequencyDict)
+    {
+        this.freqDict = frequencyDict;
+    }
+
+    public void add(Long freq)
+    {
+        if (freqDict.containsKey(freq)) {
+            freqDict.put(freq, freqDict.get(freq) + 1);
+        }
+        else {
+            freqDict.put(freq, 1L);
+        }
+    }
+
+    public void merge(NDVEstimatorState otherState)
+    {
+        Map<Long, Long> otherFreqDict = otherState.getFreqDict();
+        for (Long value : otherFreqDict.keySet()) {
+            if (freqDict.containsKey(value)) {
+                freqDict.put(value, freqDict.get(value) + otherFreqDict.get(value));
+            }
+            else {
+                freqDict.put(value, otherFreqDict.get(value));
+            }
+        }
+    }
+
+    private long computeBirthdayProblemProbability(long d)
+    {
+        double lowerBoundProbability = 0.1;
+        long i = d;
+        long lowerBound;
+        while (true) {
+            lowerBound = d + i;
+            double multiplied = 1;
+            for (int j : IntStream.range(0, (int) d).toArray()) {
+                multiplied *= (double) (lowerBound - j) / lowerBound;
+            }
+            if (multiplied > lowerBoundProbability) {
+                break;
+            }
+            i += 1;
+        }
+        return lowerBound;
+    }
+
+    public long estimate()
```

Comment:
It would be useful either in a javadoc comment on the class or on this method to include a reference to the paper and/or algorithm being used so those in the future can understand where all of the logic comes from. 

Personally, I would put the info on the class-level javadoc to explain the entire high-level usage of this estimator.